### PR TITLE
games-roguelike/dwarf-fortress: add version 0.47.04, bugfix

### PR DESCRIPTION
**Pull request by sveneusewig**
The bug has already been confirmed but has not yet been fixed.
The changes for the bug fix were posted by the user Cheaterman
https://www.bay12games.com/dwarves/mantisbt/view.php?id=11564

Signed-off-by: Sven Eusewig <sveneusewig@yahoo.de>
Bug: https://bugs.gentoo.org/729002
Package-Manager: Portage-3.0.4, Repoman-3.0.1

### CODE REVIEW

**Comment by Cheaterman** [comment]
Thank you :-) :+1:

**Comment by sveneusewig** [comment]
@Cheaterman , @chewi 
Who else needs to know?

File: games-roguelike/dwarf-fortress/dwarf-fortress-0.47.04.ebuild [hunk 1]
```diff
@@ -43,6 +43,9 @@ RESTRICT="strip"
 src_prepare() {
 	rm -f libs/*.so* || die
```
Comment (DarthGandalf) [inline games-roguelike/dwarf-fortress/dwarf-fortress-0.47.04.ebuild:44]: Please remove `-f`

File: games-roguelike/dwarf-fortress/dwarf-fortress-0.47.04.ebuild [hunk 2]
```diff
@@ -1,12 +1,12 @@
 # Copyright 1999-2020 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=6
+EAPI=7
```
Comment (DarthGandalf) [inline games-roguelike/dwarf-fortress/dwarf-fortress-0.47.04.ebuild:4]: In EAPI 7 `BDEPEND` should be split from `DEPEND`

**Review by DarthGandalf — COMMENTED** [review]
This also closes https://bugs.gentoo.org/717752 so please add it to commit message

File: games-roguelike/dwarf-fortress/dwarf-fortress-0.47.04.ebuild (hunk 2 above)
Comment (sveneusewig) [reply games-roguelike/dwarf-fortress/dwarf-fortress-0.47.04.ebuild:4]: I add this: `BDEPEND="virtual/pkgconfig"` and remove `"virtual/pkgconfig"` from `DEPEND`.

File: games-roguelike/dwarf-fortress/dwarf-fortress-0.47.04.ebuild (hunk 1 above)
Comment (sveneusewig) [reply games-roguelike/dwarf-fortress/dwarf-fortress-0.47.04.ebuild:44]: Ok.

File: games-roguelike/dwarf-fortress/dwarf-fortress-0.47.04.ebuild [hunk 3]
```diff
@@ -31,8 +31,8 @@ RDEPEND="media-libs/glew:0
 DEPEND="${RDEPEND}
 	media-libs/libsndfile
 	media-libs/openal
-	sys-libs/ncurses-compat:5[unicode]
-	virtual/pkgconfig"
+	sys-libs/ncurses-compat:5[unicode]"
```
Comment (chewi) [inline games-roguelike/dwarf-fortress/dwarf-fortress-0.47.04.ebuild:34]: I know you haven't changed this but `ncurses-compat` in `DEPEND` doesn't make any sense. This package does not install any headers. I can see that the sources are hardcoded to load `libncurses.so.5`. This is really bad if you're building with the headers from `libncurses.so.6`. I can see that it falls back to `libncurses.so` but that's no use if it finds `.so.5` first. The fact that it builds means it must be compatible with `.so.6` so I guess you should patch `renderer_curses.cpp` accordingly. This whole thing is really horrible though.

**Review by chewi — CHANGES_REQUESTED** [review]
This isn't new but I really don't like how the launcher for this game just copies the entire thing into `HOME`. Is this really the only way it can work? I don't have the energy to argue with upstream about it.

**Comment by chewi** [comment]
> This isn't new but I really don't like how the launcher for this game just copies the entire thing into `HOME`. Is this really the only way it can work? I don't have the energy to argue with upstream about it.

Ah sorry, I now realise this launcher script is our own. :fearful: Presumably it was done like this for a reason though. Please find out whether it's still necessary.

File: games-roguelike/dwarf-fortress/dwarf-fortress-0.47.04.ebuild [hunk 4]
```diff
@@ -41,8 +41,11 @@ QA_PREBUILT="${gamesdir#/}/libs/Dwarf_Fortress"
 RESTRICT="strip"
 
 src_prepare() {
-	rm -f libs/*.so* || die
+	rm libs/*.so* || die
 	sed -i -e '1i#include <cmath>' g_src/ttf_manager.cpp || die
+
+	eapply "${FILESDIR}/${P}-segfault-fix-729002.patch"
```
[the remark this answers is not in the document]
Comment (sveneusewig) [reply games-roguelike/dwarf-fortress/dwarf-fortress-0.47.04.ebuild:47]: Ok. I remove this there.

File: games-roguelike/dwarf-fortress/dwarf-fortress-0.47.04.ebuild (hunk 3 above)
Comment (sveneusewig) [reply games-roguelike/dwarf-fortress/dwarf-fortress-0.47.04.ebuild:34]: Let's see what I can find. In the worst case it looks like this:
https://i.imgur.com/Kx4NvI4.gif

**Comment by frayment** [comment]
You may want to update this to 0.47.05 now that it's out. I've tested your patch in the new version with a custom libgraphic build and it still works.

**Comment by sveneusewig** [comment]
I'm making a new pull request.

**Pull request closed without merging**